app/handler/unsubscribe_generator.py

```diff
@@ -1,34 +1,90 @@
+import urllib
 from email.message import Message
 
 from app.email import headers
 from app.email_utils import add_or_replace_header
 from app.handler.unsubscribe_encoder import (
     UnsubscribeEncoder,
     UnsubscribeAction,
+    UnsubscribeData,
+    UnsubscribeOriginalData,
 )
-from app.models import Alias, Contact
+from app.models import Alias, Contact, UnsubscribeBehaviourEnum
 
 
 class UnsubscribeGenerator:
-    def add_header_to_message(
+    def _generate_header_with_sl_behaviour(
         self, alias: Alias, contact: Contact, message: Message
     ) -> Message:
         """
         Add List-Unsubscribe header
         """
         user = alias.user
         if user.one_click_unsubscribe_block_sender:
-            unsub_link = UnsubscribeEncoder.encode(
-                UnsubscribeAction.DisableContact, contact.id
-            )
+            unsub = UnsubscribeData(UnsubscribeAction.DisableContact, contact.id)
         else:
-            unsub_link = UnsubscribeEncoder.encode(
-                UnsubscribeAction.DisableAlias, alias.id
+            unsub = UnsubscribeData(UnsubscribeAction.DisableAlias, alias.id)
+        return self._add_unsubscribe_header(message, unsub)
+
+    def _generate_header_with_original_behaviour(
+        self, alias: Alias, message: Message
+    ) -> Message:
```

Comment:
Do you think we can add some doc for some steps in this method? It's a little bit hard to know what's going on in the implementation.

Reply:
Will do.

Reply:
Done